evm_arithmetization/src/generation/prover_input.rs

```diff
@@ -364,12 +375,12 @@ impl<F: RichField> GenerationState<F> {
             ));
         };
 
-        if let Some(ctx_jumpdest_table) = jumpdest_table.get_mut(&context)
+        if let Some(ctx_jumpdest_table) = (*jumpdest_table).get_mut(&context)
             && let Some(next_jumpdest_address) = ctx_jumpdest_table.pop()
         {
             Ok((next_jumpdest_address + 1).into())
         } else {
-            jumpdest_table.remove(&context);
+            (*jumpdest_table).remove(&context);
```

Comment:
unneeded changes?

Reply:
It is needed now because method calls bind stronger than dereferencing, [precedence table](https://doc.rust-lang.org/reference/expressions.html#expression-precedence).

Reply:
Hmm I definitely am no expert in precedence table ordering, but reverting your changes definitely works because we're having simple wrappers with a Deref impl.
You can try it for yourself:

```rust
        if let Some(ctx_jumpdest_table) = jumpdest_table.get_mut(&context)
            && let Some(next_jumpdest_address) = ctx_jumpdest_table.pop()
        {
            Ok((next_jumpdest_address + 1).into())
        } else {
            jumpdest_table.remove(&context);
            Ok(U256::zero())
        }
```

Reply:
I stand corrected: it is not needed, because explicit dereferencing is not needed.